src/main/java/org/owasp/validator/html/Policy.java

```diff
@@ -365,7 +370,7 @@ protected static Element getTopLevelElement(InputSource source, Callable<InputSo
         // Track whether an exception was ever thrown while processing policy file
         Exception thrownException = null;
         try {
-            return getDocumentElementFromSource(source, true);
+            return getDocumentElementFromSource(source, validateSchema);
```

Comment:
The log content in this method never gets hit.   I think your tests are great forms of validation, but we're missing a required logic path in the class updates.

To verify this, I've cloned this content locally and made the following adjustments:
 - Renamed `AntiSammyMalformedPolicy` to `AntiSamyMalformedPolicyTest`

**Captured Output**
```
$ mvn test -B  -Dtest=AntiSamyMalformedPolicyTest.java
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building OWASP AntiSamy 1.6.1
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-bytecode-version) @ antisamy ---
[INFO] Adding ignore: module-info
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-jdk-version) @ antisamy ---
[INFO]
[INFO] --- maven-resources-plugin:3.2.0:resources (default-resources) @ antisamy ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 19 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ antisamy ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:3.2.0:testResources (default-testResources) @ antisamy ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 384 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ antisamy ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ antisamy ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.owasp.validator.html.test.AntiSamyMalformedPolicyTest
[main] INFO org.owasp.validator.html.Policy - Attempting to load AntiSamy policy from an input stream.
[main] WARN org.owasp.validator.html.Policy - Attempting to load AntiSamy policy from an input stream.
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.319 s - in org.owasp.validator.html.test.AntiSa
myMalformedPolicyTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.194 s
[INFO] Finished at: 2021-03-14T10:21:54-05:00
[INFO] Final Memory: 13M/318M
[INFO] ------------------------------------------------------------------------
```

Running the full build, there is 1 instance in the logs that include the text "Invalid AntiSamy policy file".  That log is coming from line 460 of Policy.java, and is a result of `PolicyTest.testSchemaValidationToggleWithInclude`.

Reply:
I should clarify,  when running the AntiSamyMalformedPolcyTest in isolation, I was expecting to see at least one record of `"Invalid AntiSamy policy file: "`  (line 381)

Reply:
@davewichers   Do you care about this difference in logging?  I may be misinterpreting a previous, and should have clarified sooner.

Reply:
In that case, an exception is been thrown:

        try {
            Policy.getInstance(stream);
            fail("Malformed XML, PolicyException expected ");
        **} catch (PolicyException e) {
            // Expected behavior
        }**

I can add an output to the test case, but as the test is working as expected, no output is needed.

Reply:
I agree with you @gerardocanedo .  The change you've made will meet all of the behavioral expectations of the API contract with the Policy class. 

My focus is that the logic flow is now different, which results in a loss of logging or dead code in the baseline.  You've stated that it is intentional (something I was previously unclear on), and I will admit that I may have taken the comment from Dave a bit too literal in my assessments.   If that is the case, then I do apologize.

At the end of the day, this get ESAPI what is needed for the project to move forward.  It comes down to whether the loss of the logging is acceptable to the AntiSamy maintainers.  Assuming it is, the next question would be whether the dead code should be removed in this review to avoid future confusion.

I suppose it's fair to say that since this is already merged, that no further work should be required, and overall the continuation of this discussion  would be moot. 

I am grateful for the feedback and responsiveness to my comments, I  really do appreciate how quickly you picked this up and rolled the adjustment out.

Reply:
Thank you for all the review. 
It's my first contribution, so I'm learning a lot about this and what's expected.
I agree about dead code. I'll remove it. I was a little afraid about it.